lib/finch/pool.ex

```diff
@@ -25,11 +23,19 @@ defmodule Finch.Pool do
     receive_timeout = Keyword.get(opts, :receive_timeout, 15000)
 
     NimblePool.checkout!(pool, :checkout, fn {conn, pool} ->
+      # do we need this try catch?
```

Comment:
It depends. It may not be necessary as long as we can close the connection from a process that does not own it. You can test it by doing something like this in IEx:

```
conn = Conn.connect(conn)
spawn fn -> Conn.close(conn) end
```

If the connection is closed afterwards, then you can remove the try/catch and the pool will close it for you. If you need to close it in the caller though, then you need to keep it. Please keep me posted as I should update NimblePool's readme too.

Reply:
It appears that you do not need to be the owner to close a Conn:
```
iex(1)> alias Finch.Conn                                      
Finch.Conn
iex(2)> conn = Conn.new(:https, "github.com", 443, [], self())
%{
  host: "github.com",
  mint: nil,
  opts: [],
  parent: #PID<0.311.0>,
  port: 443,
  scheme: :https
}
iex(3)> conn = Conn.connect(conn)                             
%{
  host: "github.com",
  mint: %Mint.HTTP1{
    buffer: "",
    host: "github.com",
    mode: :active,
    port: 443,
    private: %{},
    request: nil,
    requests: {[], []},
    scheme_as_string: "https",
    socket: {:sslsocket, {:gen_tcp, #Port<0.11>, :tls_connection, :undefined},
     [#PID<0.319.0>, #PID<0.318.0>]},
    state: :open,
    transport: Mint.Core.Transport.SSL
  },
  opts: [],
  parent: #PID<0.311.0>,
  port: 443,
  scheme: :https
}
iex(4)> spawn fn -> Conn.close(conn) |> IO.inspect end        
#PID<0.321.0>
%{      
  host: "github.com",
  mint: %Mint.HTTP1{
    buffer: "",
    host: "github.com",
    mode: :active,
    port: 443,
    private: %{},
    request: nil,
    requests: {[], []},
    scheme_as_string: "https",
    socket: {:sslsocket, {:gen_tcp, #Port<0.11>, :tls_connection, :undefined},
     [#PID<0.319.0>, #PID<0.318.0>]},
    state: :closed,
    transport: Mint.Core.Transport.SSL
  },
  opts: [],
  parent: #PID<0.311.0>,
  port: 443,
  scheme: :https
}
```

Reply:
So I think we are good to go without it!